reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -379,13 +379,27 @@ class StaticHtmlReporter : Reporter() {
         h2 { +"Index" }
 
         ul {
-            reportTableModel.evaluatorIssues?.let {
-                li { a("#policy-violation-summary") { +"Rule Violation Summary (${it.size} violations)" } }
+            reportTableModel.evaluatorIssues?.let { ruleViolations ->
+                val issues = ruleViolations.groupBy { it.severity }
+                val errorCount = issues[Severity.ERROR].orEmpty().count()
+                val warningCount = issues[Severity.WARNING].orEmpty().count()
+                val hintCount = issues[Severity.HINT].orEmpty().count()
```

Comment:
There is also this option:

```kotlin
        val counts = issues.groupingBy { it.severity }.eachCount()
        val errors = counts[Severity.ERROR] ?: 0
```

I was curious which performs better and found that the variant with `groupBy` is slightly faster up to 1,000,000 elements in the list, but above that the performance breaks down and at 10,000,000 elements it is 20 times slower than the variant with `groupingBy`.

Reply:
Interesting! Assuming that we mostly have < 1,000,000 issues, I guess we stick to the current solution 😉

Reply:
Yes, I think if you have so many issues you can wait a few more seconds for the report ;-)
Btw, the solution you had in the first version of the PR is 2x faster for a reasonable number of issues (<1000), but we are talking about <1ms here.

Reply:
Thanks for taking the time to benchmark that one, too. Good to know I can still write performant code 😆